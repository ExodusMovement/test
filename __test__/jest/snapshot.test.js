it.skip('simple', () => {
  expect(10).toMatchSnapshot()
  expect(null).toMatchSnapshot()
  expect().toMatchSnapshot()
  expect([]).toMatchSnapshot()
  expect(/xx/).toMatchSnapshot()
  expect(Infinity).toMatchSnapshot()
  expect(false).toMatchSnapshot()
  expect(true).toMatchSnapshot()
  expect({}).toMatchSnapshot()
})

it('complex', () => {
  expect([10]).toMatchSnapshot()
  expect([{ a: 10 }]).toMatchSnapshot()
  expect({ a: 10 }).toMatchSnapshot()
  expect({ a: 10, b: 20 }).toMatchSnapshot()
  expect(Buffer.from('')).toMatchSnapshot()
})

it('simple inline', () => {
  expect(42).toMatchInlineSnapshot(`42`)
  expect({}).toMatchInlineSnapshot(`{}`)
})

const TEST_ONE = { a: 20, d: Buffer.from('foo'), b: [1, 2, 'bar', 5], e: { foo: 'bar' } }
// eslint-disable-next-line no-sparse-arrays
const TEST_TWO = { ['__proto__']: [], b: [1, 2, , , 5], e: { foo: 'bar' }, f: -Infinity }
const TEST_THREE = [new Error('foo'), new TypeError('bar'), new Uint16Array(4, 2, 1_000_000), null]

// Test names repeat on a purpose!

it('test A', () => {
  expect(TEST_ONE).toMatchSnapshot()
})

it('test B', () => {
  expect(TEST_TWO).toMatchSnapshot()
})

it('test B', () => {
  expect(TEST_THREE).toMatchSnapshot()
})

// Repeat name
it('test B', () => {
  expect({ x: 1337 }).toMatchSnapshot()
})

describe('nested test', () => {
  it('test A', () => {
    expect(TEST_TWO).toMatchSnapshot()
  })

  it('test A', () => {
    expect(TEST_THREE).toMatchSnapshot()
  })

  it('nested test one', () => {
    expect(TEST_ONE).toMatchSnapshot()
  })
})

it('test one, inline', () => {
  // eslint-disable-next-line unicorn/template-indent
  expect(TEST_ONE).toMatchInlineSnapshot(`
{
  "a": 20,
  "b": [
    1,
    2,
    "bar",
    5,
  ],
  "d": {
    "data": [
      102,
      111,
      111,
    ],
    "type": "Buffer",
  },
  "e": {
    "foo": "bar",
  },
}
`)
})

it('errors', () => {
  expect(() => {
    throw new TypeError('Whoops\nI\nFailed')
  }).toThrowErrorMatchingSnapshot()

  expect(() => {
    throw new RangeError('Out of something')
  }).toThrowErrorMatchingInlineSnapshot('"Out of something"')
})
