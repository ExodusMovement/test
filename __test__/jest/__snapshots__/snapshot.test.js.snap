// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async errors 1`] = `"slow but nah"`;

exports[`complex 1`] = `
[
  10,
]
`;

exports[`complex 2`] = `
[
  {
    "a": 10,
  },
]
`;

exports[`complex 3`] = `
{
  "a": 10,
}
`;

exports[`complex 4`] = `
{
  "a": 10,
  "b": 20,
}
`;

exports[`complex 5`] = `
{
  "data": [],
  "type": "Buffer",
}
`;

exports[`errors 1`] = `
"Whoops
I
Failed"
`;

exports[`mixed 1`] = `true`;

exports[`mixed 2`] = `
[
  1,
  2,
  3,
]
`;

exports[`mixed 3`] = `
{
  "foo": "bar",
}
`;

exports[`mixed 4`] = `43`;

exports[`mixed 5`] = `{}`;

exports[`mixed 6`] = `[]`;

exports[`mixed 7`] = `
[
  5,
  4,
  3,
]
`;

exports[`mixed 8`] = `[]`;

exports[`mixed 9`] = `false`;

exports[`mixed 10`] = `41`;

exports[`mixed 11`] = `
{
  "bar": "buz",
}
`;

exports[`mixed 12`] = `{}`;

exports[`nested test nested test one 1`] = `
{
  "a": 20,
  "b": [
    1,
    2,
    "bar",
    5,
  ],
  "d": {
    "data": [
      102,
      111,
      111,
    ],
    "type": "Buffer",
  },
  "e": {
    "foo": "bar",
  },
}
`;

exports[`nested test test A 1`] = `
{
  "__proto__": [],
  "b": [
    1,
    2,
    ,
    ,
    5,
  ],
  "e": {
    "foo": "bar",
  },
  "f": -Infinity,
}
`;

exports[`nested test test A 2`] = `
[
  [Error: ?!],
  [TypeError: bar],
  Uint16Array [
    4,
    2,
    65123,
  ],
  null,
]
`;

exports[`simple 1`] = `10`;

exports[`simple 2`] = `null`;

exports[`simple 3`] = `undefined`;

exports[`simple 4`] = `[]`;

exports[`simple 5`] = `/xx/`;

exports[`simple 6`] = `Infinity`;

exports[`simple 7`] = `false`;

exports[`simple 8`] = `true`;

exports[`simple 9`] = `{}`;

exports[`simple 10`] = `/hello/`;

exports[`simple 11`] = `true`;

exports[`simple 12`] = `NaN`;

exports[`simple 13`] = `{}`;

exports[`simple 14`] = `42`;

exports[`simple 15`] = `[]`;

exports[`simple 16`] = `-Infinity`;

exports[`test A 1`] = `
{
  "a": 20,
  "b": [
    1,
    2,
    "bar",
    5,
  ],
  "d": {
    "data": [
      102,
      111,
      111,
    ],
    "type": "Buffer",
  },
  "e": {
    "foo": "bar",
  },
}
`;

exports[`test B 1`] = `
{
  "__proto__": [],
  "b": [
    1,
    2,
    ,
    ,
    5,
  ],
  "e": {
    "foo": "bar",
  },
  "f": -Infinity,
}
`;

exports[`test B 2`] = `
[
  [Error: ?!],
  [TypeError: bar],
  Uint16Array [
    4,
    2,
    65123,
  ],
  null,
]
`;

exports[`test B 3`] = `
{
  "x": 1337,
}
`;
